server/src/main/java/com/linecorp/centraldogma/server/internal/metadata/MetadataService.java

```diff
@@ -0,0 +1,853 @@
+                                           TOKEN_JSON,
+                                           asJsonArray(new RemoveOperation(appIdPath),
+                                                       new RemoveIfExistsOperation(secretPath)));
+                                   return WithRevision.of(change, tokens.revision());
+                               }))
+        );
+    }
+
+    /**
+     * Activates the {@link Token} of the specified {@code appId}.
+     */
+    public CompletionStage<Revision> activateToken(Author author, String appId) {
+        requireNonNull(author, "author");
+        requireNonNull(appId, "appId");
+
+        return tokenRepo.push(INTERNAL_PROJECT_NAME, TOKEN_REPO, author,
+                              "Enable the token: '" + appId,
+                              () -> tokenRepo
+                                      .fetch(INTERNAL_PROJECT_NAME, TOKEN_REPO, TOKEN_JSON)
+                                      .thenApply(tokens -> {
+                                          final Token token = tokens.object().get(appId);
+                                          final JsonPointer removalPath =
+                                                  JsonPointer.compile("/appIds/" + appId + "/deactivation");
+                                          final JsonPointer secretPath =
+                                                  JsonPointer.compile("/secrets/" + token.secret());
+                                          final Change<JsonNode> change = Change.ofJsonPatch(
+                                                  TOKEN_JSON,
+                                                  asJsonArray(new RemoveOperation(removalPath),
+                                                              new AddOperation(secretPath,
+                                                                               Jackson.valueToTree(appId))));
+                                          return WithRevision.of(change, tokens.revision());
+                                      })
+        );
+    }
+
+    /**
+     * Deactivates the {@link Token} of the specified {@code appId}.
+     */
+    public CompletionStage<Revision> deactivateToken(Author author, String appId) {
+        requireNonNull(author, "author");
+        requireNonNull(appId, "appId");
+
+        return tokenRepo.push(INTERNAL_PROJECT_NAME, TOKEN_REPO, author,
+                              "Disable the token: '" + appId,
+                              () -> tokenRepo
+                                      .fetch(INTERNAL_PROJECT_NAME, TOKEN_REPO, TOKEN_JSON)
+                                      .thenApply(tokens -> {
+                                          final Token token = tokens.object().get(appId);
+                                          final JsonPointer removalPath =
+                                                  JsonPointer.compile("/appIds/" + appId + "/deactivation");
```

Comment:
Same for `appId` - It would be great if we have a single place that validates all `appId` strings.

Reply:
Added validation for `appId` and `secret` to the constructor of `Token` class. I think it's enough to use `Util.validateFileName()` for these cases. Please let me know if you have any concern about that.